server/multi_server_integration_test.go

```
@@ -0,0 +1,362 @@
+type MultiServerIntegrationTestSuite struct {
+	suite.Suite
+	containerA *matrixtest.Container
+	containerB *matrixtest.Container
+	plugin     *Plugin
+	api        *plugintest.API
+}
+
+func (suite *MultiServerIntegrationTestSuite) SetupSuite() {
+	// Two independent homeservers with distinct domains (each gets its own
+	// dynamically-assigned port from testcontainers).
+	suite.containerA = matrixtest.StartMatrixContainer(suite.T(), matrixtest.MatrixTestConfig{
```

golangci-lint (2.12.2)

[high] G101: Potential hardcoded credentials

(gosec)